include/rmm/cuda_stream_view.hpp

```diff
@@ -33,19 +33,31 @@ namespace rmm {
  */
 class cuda_stream_view {
  public:
-  constexpr cuda_stream_view()                                   = default;
-  constexpr cuda_stream_view(cuda_stream_view const&)            = default;
-  constexpr cuda_stream_view(cuda_stream_view&&)                 = default;
+  constexpr cuda_stream_view()                        = default;
+  constexpr cuda_stream_view(cuda_stream_view const&) = default;  ///< Default copy constructor
+  constexpr cuda_stream_view(cuda_stream_view&&)      = default;  ///< Default move constructor
+  /**
+   * @brief Default copy assignment operator
+   *
+   * @return cuda_stream_view& Reference to the assigned object
+   */
   constexpr cuda_stream_view& operator=(cuda_stream_view const&) = default;
-  constexpr cuda_stream_view& operator=(cuda_stream_view&&)      = default;
-  ~cuda_stream_view()                                            = default;
+  /**
+   * @brief Default move assignment operator
+   *
+   * @return cuda_stream_view& Reference to the assigned object
+   */
+  constexpr cuda_stream_view& operator=(cuda_stream_view&&) = default;
+  ~cuda_stream_view()                                       = default;
```

Comment:
I see you didn't write an obvious destructor doc? So maybe warnings are not the motivation for these?